AEFeinstein/Swadge-IDF-5.0
# Added Pong Mode

### Description

The main change is adding the Pong and Main Menu modes.

Separate, reusable geometry and vector math utility functions were added.

A bug was fixed which would chirp the speakers when the emulator started.

The system was also heavily documented.

I'm sure there are unlisted tweaks as well, but I'm the only one working on the system for now, so it's cool.

### Test Instructions

Make sure the menu works and the mode enters and exits cleanly.

Play pong. Make sure to test different input schemes:
1. Button
1. Touch
1. Tilt

Test the CPU difficulties:
1. Easy should sweep up and down
1. Medium should track the ball slowly
1. Hard should never miss

### Ticket Links

n/a

### Readiness Checklist
